### ISSUE TITLE
perform thicc refactor on Calcdex component logic

### ISSUE THREAD

**Issue by doshidak** [doshidak/showdex#96]
`Calcdex` and its children components (e.g., `PlayerCalc`, `PokeCalc`, etc.) have a lot of component-specific logic and prop passing, remnants of the initial implementation, i.e., v0.1.0.

All this to say that it's super gross and confusing, especially with `useCalcdex()`, which was my initial refactoring attempt at consolidating the props and logic, but still requires prop passing cause the hook is not a `React.Context`, so the logic would re-fire for each component that uses the hook.

Basically, move all the specialized functions like `handlePokemonChange()` in `PokeCalc` and `applyPreset()` in `PokeInfo` into a `CalcdexProvider`, then remove all the prop passing in the children components by using a `React.Context` consumer as a hook (via `React.useContext()`).

`usePresets()` should also be used within `CalcdexProvider`, with its return values accessible by consumers.

Component implementations should focus on the UI/UX, not business logic.
